tony-history-server/app/controllers/JobsMetadataPageController.java

```diff
@@ -38,13 +39,14 @@ public Result index() {
     }
 
     List<JobMetadata> listOfMetadata = new ArrayList<>();
-    String tonyHistoryFolder = config.getString("tony.historyFolder");
+    Path tonyHistoryFolder = new Path(config.getString("tony.historyFolder"));
```

Comment:
Please use `TonyConfigurationKeys.TONY_HISTORY_LOCATION`. Same elsewhere in the code. This also helps remove inconsistencies such as using `tony.historyFolder` in some places and `tony.history.location` in others.

Reply:
A few clarifications:

> make Tony AM read from tony-site.xml

TonYClient would read from `tony-site.xml`, apply passed in args, and then write out `tony-final.xml`, which is what the AM uses (no changes in AM).

> write some intermediate script to extract the location from tony-site.xml and inject it into application.conf

`startTHS.sh` can do this and set `-Dprop1=val1` Java system properties that will override what's in `application.conf`. For inspiration, take a look at what Dr. Elephant does here: https://github.com/linkedin/dr-elephant/blob/master/scripts/start.sh

> I don't think that would make the project more maintainable as we're introducing an extra script to maintain, and the location of the job history folder doesn't (and shouldn't) change once we deploy this to prod, so the benefit of having a single source of truth (tony-site.xml) is also very minimal.

Right now, it's just one variable, but in the future, there might be more variables that we need to be consistent between client and server. We should code using best practices from the beginning to save us technical debt in the future.

Reply:
Noted. Will work on this in the upcoming PR. @pdtran3k6